engine/src/test/java/net/jqwik/api/EMailsTests.java

```diff
@@ -29,54 +33,27 @@ void validLengthAfterAt(@ForAll("emails") String email){
 	}
 
 	@Property
-	void validSignsBeforeAt(@ForAll("emails") String email){
+	void validSignsBeforeAtUnquoted(@ForAll("emails") String email){
 		String localPart = getLocalPartOfEmail(email);
-		if(isQuoted(localPart)){
-			localPart = localPart.substring(1, localPart.length() - 1);
-			localPart = localPart.replace( " ", "");
-			localPart = localPart.replace( "(", "");
-			localPart = localPart.replace( ")", "");
-			localPart = localPart.replace( ",", "");
-			localPart = localPart.replace( ":", "");
-			localPart = localPart.replace( ";", "");
-			localPart = localPart.replace( "<", "");
-			localPart = localPart.replace( ">", "");
-			localPart = localPart.replace( "@", "");
-			localPart = localPart.replace( "[", "");
-			localPart = localPart.replace( "\\\\", "");
-			localPart = localPart.replace( "]", "");
-			localPart = localPart.replace( "\\\"", "");
-		}
-		for(char c = 'a'; c <= 'z'; c++){
-			localPart = localPart.replace(c + "", "");
-		}
-		for(char c = 'A'; c <= 'Z'; c++){
-			localPart = localPart.replace(c + "", "");
-		}
-		for(char c = '0'; c <= '9'; c++){
-			localPart = localPart.replace(c + "", "");
-		}
-		localPart = localPart.replace( "!", "");
-		localPart = localPart.replace( "#", "");
-		localPart = localPart.replace( "$", "");
-		localPart = localPart.replace( "%", "");
-		localPart = localPart.replace( "&", "");
-		localPart = localPart.replace( "'", "");
-		localPart = localPart.replace( "*", "");
-		localPart = localPart.replace( "+", "");
-		localPart = localPart.replace( "-", "");
-		localPart = localPart.replace( "/", "");
-		localPart = localPart.replace( "=", "");
-		localPart = localPart.replace( "?", "");
-		localPart = localPart.replace( "^", "");
-		localPart = localPart.replace( "_", "");
-		localPart = localPart.replace( "`", "");
-		localPart = localPart.replace( "{", "");
-		localPart = localPart.replace( "|", "");
-		localPart = localPart.replace( "}", "");
-		localPart = localPart.replace( "~", "");
-		localPart = localPart.replace( ".", "");
-		Assertions.assertThat(localPart).isEqualTo("");
+		Assume.that(!isQuoted(localPart));
+		Assertions.assertThat(localPart.chars()).allMatch(c -> ALLOWED_CHARS_LOCALPART_UNQUOTED.contains(((char) c.intValue()) + ""));
+	}
+
+	@Property
+	void validSignsBeforeAtQuoted(@ForAll("emails") String email){
+		String localPart = getLocalPartOfEmail(email);
+		Assume.that(isQuoted(localPart));
+		Assertions.assertThat(localPart.chars()).allMatch(c -> ALLOWED_CHARS_LOCALPART_QUOTED.contains(((char) c.intValue()) + ""));
+	}
+
+	@Property
+	void validUseOfQuotedBackslashAndQuotationMarks(@ForAll("emails") String email){
+		String localPart = getLocalPartOfEmail(email);
+		Assume.that(isQuoted(localPart));
+		localPart = localPart.substring(1, localPart.length() - 1);
+		localPart = localPart.replace("\\\"", "").replace("\\\\", "");
+		Assertions.assertThat(localPart).doesNotContain("\\");
+		Assertions.assertThat(localPart).doesNotContain("\"");
```

Comment:
Is this replace method okay? Because I don't know how I can check that localPart only contains \\" instead of " and \\\\ instead of \\ only with Assertions.